### PR TITLE
8345294: test/jdk/java/lang/Thread/virtual/RetryMonitorEnterWhenPinned.java timeout with JTREG_TEST_THREAD_FACTORY=Virtual

### PR DESCRIPTION
**Pull request by DougLea**
Accommodate compensations when checking for missed signals; add a better test for this.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345294](https://bugs.openjdk.org/browse/JDK-8345294): test/jdk/java/lang/Thread/virtual/RetryMonitorEnterWhenPinned.java timeout with JTREG_TEST_THREAD_FACTORY=Virtual (**Bug** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22543/head:pull/22543` \
`$ git checkout pull/22543`

Update a local copy of the PR: \
`$ git checkout pull/22543` \
`$ git pull https://git.openjdk.org/jdk.git pull/22543/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22543`

View PR using the GUI difftool: \
`$ git pr show -t 22543`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22543.diff">https://git.openjdk.org/jdk/pull/22543.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22543#issuecomment-2517527339)
</details>

### CODE REVIEW

File: test/jdk/java/lang/Thread/virtual/Starvation.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/* @test
+ * @library /test/lib
```
Comment (viktorklang-ora) [inline test/jdk/java/lang/Thread/virtual/Starvation.java:2]:
```suggestion
 * @bug 8345294
 * @library /test/lib
```

**Comment by viktorklang-ora** [comment]
Thanks for the fix, Doug, and thanks for the reproducer test, @AlanBateman!

Having a look at the patch now 👍

File: test/jdk/java/lang/Thread/virtual/Starvation.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/* @test
```
[the remark this answers is not in the document]
Comment (DougLea) [reply test/jdk/java/lang/Thread/virtual/Starvation.java:1]: Oops. Thanks.

[the remark this answers is not in the document]
Comment (viktorklang-ora) [reply test/jdk/java/lang/Thread/virtual/Starvation.java:24]: Np :)

File: test/jdk/java/lang/Thread/virtual/Starvation.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        var threads = new ArrayList<Thread>();
+        Object lock = new Object();
+        synchronized (lock) {
+            for (int i = 0; i < nprocs - 1; i++) {
+                var started = new CountDownLatch(1);
+                Thread thread = Thread.startVirtualThread(() -> {
+                    started.countDown();
+                    VThreadPinner.runPinned(() -> {
+                        synchronized (lock) {
+                        }
```
Comment (viktorklang-ora) [inline test/jdk/java/lang/Thread/virtual/Starvation.java:72]: @AlanBateman / @DougLea Is there any risk that this gets elided?

**Review by AlanBateman — APPROVED** [review]
At some point we can add @Summary to the test as reminder of its purpose. Otherwise this looks good, and happy this was caught.

**Pull request closed without merging**